### PR TITLE
Removes original recipient from the 'To' and 'Cc' fields when replying to  an email.

### PR DESCRIPTION
**Pull request by StCyr**
this PR can also be used to test https://github.com/nextcloud/mail/pull/2448

Signed-off-by: Cyrille Bollu <cyrpub@bollu.be>

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> this PR can also be used to test #2448

No, you opened a PR with a branch on this repo. You'd have to use a branch on your fork to test if those PRs also trigger the actions :)

File: src/components/NewMessageDetail.vue [hunk 1]
```diff
@@ -82,11 +82,22 @@ export default {
 					subject = buildForwardSubject(message.subject)
 					msgTo = []
 				}
+				
+				// Remove original recipient from 'To' and 'Cc' fields
+				// This only works if original recipient's email address is the same as the email
+				// address of the account for which the original email was received, not an alias.
+				var account = this.$store.getters.getAccount(message.accountId)
```
Comment (ChristophWurst) [inline src/components/NewMessageDetail.vue]: For replies we have the `buildRecipients` helper in `ReplyBuilder.js`. What about doing the same for forward? If we extract this little logic into an external function we can unit test it :)

Comment (StCyr) [reply src/components/NewMessageDetail.vue]: Nice :-)

I'll have a look :-)

Comment (ChristophWurst) [reply src/components/NewMessageDetail.vue]: https://github.com/nextcloud/mail/blob/master/src/ReplyBuilder.js and https://github.com/nextcloud/mail/blob/master/src/tests/unit/ReplyBuilder.spec.js are the files. You can run the js tests with `npm run test:watch` during development and see the results almost instantly!

Comment (StCyr) [reply src/components/NewMessageDetail.vue]: this utiliy doesn't seem so be used:

```
====> function "buildRecipients" not used

root@vps612142:~/mail# grep -R buildRecipients src/
src/ReplyBuilder.js:export const buildRecipients = (envelope, ownAddress) => {
src/tests/unit/ReplyBuilder.spec.js:import {buildReplyBody, buildRecipients, buildReplySubject} from '../../ReplyBuilder'
src/tests/unit/ReplyBuilder.spec.js:            const reply = buildRecipients(envelope, a)
src/tests/unit/ReplyBuilder.spec.js:            var reply = buildRecipients(envelope, b)
src/tests/unit/ReplyBuilder.spec.js:            const reply = buildRecipients(envelope, b)
src/tests/unit/ReplyBuilder.spec.js:            const reply = buildRecipients(envelope, d)
src/tests/unit/ReplyBuilder.spec.js:            const reply = buildRecipients(envelope, e)
src/tests/unit/ReplyBuilder.spec.js:            const reply = buildRecipients(envelope, me)
src/tests/unit/ReplyBuilder.spec.js:            const reply = buildRecipients(envelope, jan)
src/components/Message.vue:import {buildRecipients as buildReplyRecipients, buildReplySubject} from '../ReplyBuilder'


====> function "buildReplyRecipients" used but not defined !?!

root@vps612142:~/mail# grep -R buildReplyRecipients src/
src/components/Message.vue:import {buildRecipients as buildReplyRecipients, buildReplySubject} from '../ReplyBuilder'
src/components/Message.vue:                                     this.replyRecipient = buildReplyRecipients(message, {
root@vps612142:~/mail# vi src/ReplyBuilder.js
root@vps612142:~/mail#
```

Comment (ChristophWurst) [reply src/components/NewMessageDetail.vue]: "import {buildRecipients as buildReplyRecipients" aliased ;)

Comment (StCyr) [reply src/components/NewMessageDetail.vue]: oups by bad :-)

**Comment by StCyr** [comment]
Created a new PR here: https://github.com/nextcloud/mail/pull/2454

**Pull request closed without merging**